payjoin/tests/integration.rs

```diff
@@ -720,22 +877,31 @@ mod integration {
 
             let payjoin = payjoin.commit_outputs();
 
-            // Select receiver payjoin inputs. TODO Lock them.
-            let available_inputs = receiver.list_unspent(None, None, None, None, None).unwrap();
-            let candidate_inputs: HashMap<Amount, OutPoint> = available_inputs
-                .iter()
-                .map(|i| (i.amount, OutPoint { txid: i.txid, vout: i.vout }))
-                .collect();
-
-            let selected_outpoint = payjoin
-                .try_preserving_privacy(candidate_inputs)
-                .expect("Failed to make privacy preserving selection");
-            let selected_utxo = available_inputs
-                .iter()
-                .find(|i| i.txid == selected_outpoint.txid && i.vout == selected_outpoint.vout)
-                .unwrap();
-            let input_pair = input_pair_from_list_unspent(selected_utxo);
-            let payjoin = payjoin.contribute_inputs(vec![input_pair]).unwrap().commit_inputs();
+            let inputs = match custom_inputs {
+                Some(inputs) => inputs,
+                None => {
+                    // Select receiver payjoin inputs. TODO Lock them.
```

Comment:
I believe signing locks them. Are you concerned about a race between tests? Could you explain the rationale behind this TODO inline?

Reply:
That was an existing comment i cut+pasted without paying attention to. Not sure what the rationale was when it was initially added.